vite build error

Element Plus version
1.2.0-beta.1

OS/Browsers version
WIN 10 / Chrome Version 97.0.4690.4

Vue version
3.2.21

Reproduction Link
https://codepen.io/xnny-the-bold/pen/oNepQJw

Steps to reproduce
demo@1.0.0 build vue-tsc --noEmit && vite build node_modules/element-plus/lib/defaults.d.ts:3:57 - error TS2307: Cannot find module '../utils/config' or its corresponding type declarations. 3 install: (app: import("vue").App<any>, opts: import("../utils/config").InstallOptions) => void; ~~~~~

node_modules/element-plus/lib/index.d.ts:7:74 - error TS2307: Cannot find module '../utils/config' or its corresponding type declarations.

7 export declare const install: (app: import("vue").App<any>, opts: import("../utils/config").InstallOptions) => void; ~~~~~

Found 2 errors.

What is Expected?
Build Success

What is actually happening?
Build Error

--- a/build/types-definitions.ts
+++ b/build/types-definitions.ts
@@ -6,10 +6,9 @@ import glob from 'fast-glob'
 import { bold } from 'chalk'
 
 import { green, red, yellow } from './utils/log'
-import { buildOutput, pkgRoot, projRoot } from './utils/paths'
+import { buildOutput, epRoot, pkgRoot, projRoot } from './utils/paths'
 
 import { excludeFiles, pathRewriter } from './utils/pkg'
-import { run } from './utils/process'
 import type { SourceFile } from 'ts-morph'
 
 const TSCONFIG_PATH = path.resolve(projRoot, 'tsconfig.json')
@@ -37,16 +36,22 @@ export const generateTypesDefinitions = async () => {
   })
 
   const filePaths = excludeFiles(
-    await glob('**/*.{js,ts,vue}', {
+    await glob(['**/*.{js,ts,vue}', '!element-plus/**/*'], {
       cwd: pkgRoot,
       absolute: true,
       onlyFiles: true,
     })
   )
+  const epPaths = excludeFiles(
+    await glob('**/*.{js,ts,vue}', {
+      cwd: epRoot,
+      onlyFiles: true,
+    })
+  )
 
   const sourceFiles: SourceFile[] = []
-  await Promise.all(
-    filePaths.map(async (file) => {
+  await Promise.all([
+    ...filePaths.map(async (file) => {
       if (file.endsWith('.vue')) {
         const content = await fs.readFile(file, 'utf-8')
         const sfc = vueCompiler.parse(content)
@@ -71,12 +76,18 @@ export const generateTypesDefinitions = async () => {
           )
           sourceFiles.push(sourceFile)
         }
-      } else if (file.endsWith('.ts')) {
+      } else {
         const sourceFile = project.addSourceFileAtPath(file)
         sourceFiles.push(sourceFile)
       }
-    })
-  )
+    }),
+    ...epPaths.map(async (file) => {
+      const content = await fs.readFile(path.resolve(epRoot, file), 'utf-8')
+      sourceFiles.push(
+        project.createSourceFile(path.resolve(pkgRoot, file), content)
+      )
+    }),
+  ])
 
   const diagnostics = project.getPreEmitDiagnostics()
   console.log(project.formatDiagnosticsWithColorAndContext(diagnostics))
@@ -115,11 +126,4 @@ export const generateTypesDefinitions = async () => {
   })
 
   await Promise.all(tasks)
-
-  const epFiles = await glob('**/*', {
-    cwd: path.resolve(outDir, 'element-plus'),
-    absolute: true,
-  })
-  await run(`mv ${epFiles.join(' ')} ${outDir}`)
-  await run(`rmdir ${path.resolve(outDir, 'element-plus')}`)
 }

--- a/packages/element-plus/package.json
+++ b/packages/element-plus/package.json
@@ -55,6 +55,11 @@
     "memoize-one": "^6.0.0",
     "normalize-wheel-es": "^1.1.0"
   },
+  "devDependencies": {
+    "@types/node": "*",
+    "csstype": "^2.6.0",
+    "vue-router": "^4.0.0"
+  },
   "vetur": {
     "tags": "tags.json",
     "attributes": "attributes.json"

--- a/renovate.json5
+++ b/renovate.json5
@@ -5,5 +5,9 @@
       matchPaths: ['docs/package.json'],
       extends: ['config:js-app'],
     },
+    {
+      matchPaths: ['packages/element-plus/package.json'],
+      extends: ['config:base', ':preserveSemverRanges'],
+    },
   ],
 }

